File: tools/bench/answers/go_sonnet/s1/14_rect_lib.go
package main

import "fmt"

type rectangle struct {
	width  float64
	height float64
}

func area(r rectangle) float64 {
	return r.width * r.height
}

func perimeter(r rectangle) float64 {
	return 2 * (r.width + r.height)
}

func scale(r rectangle, factor float64) rectangle {
	return rectangle{r.width * factor, r.height * factor}
}

func main() {
	r := rectangle{3.0, 4.0}
	fmt.Println(area(r))
	fmt.Println(perimeter(r))
	fmt.Println(area(scale(r, 2.0)))
}

File: tools/bench/answers/python_sonnet/s1/14_rect_lib.py
class Rectangle:
    def __init__(self, width, height):
        self.width = float(width)
        self.height = float(height)

def area(r):
    return r.width * r.height

def perimeter(r):
    return 2 * (r.width + r.height)

def scale(r, factor):
    return Rectangle(r.width * factor, r.height * factor)

r = Rectangle(3.0, 4.0)
print(area(r))
print(perimeter(r))
print(area(scale(r, 2.0)))
